src/presentations/indexable-presentation.php

```diff
@@ -220,19 +221,49 @@ public function generate_meta_description() {
 	 * @return array The robots value.
 	 */
 	public function generate_robots() {
-		return [
+		$robots = [
 			'index'  => ( $this->model->is_robots_noindex === true ) ? 'noindex' : 'index',
 			'follow' => ( $this->model->is_robots_nofollow === true ) ? 'nofollow' : 'follow',
 		];
+
+		// Don't return defaults, so, don't return index, follow.
+		if ( $this->model->is_robots_noindex === false &&
+			 $this->model->is_robots_nofollow === false ) {
+			return [];
+		}
+
+		return $robots;
 	}
 
 	/**
-	 * Generates the googlebot value.
+	 * Generates the robots value for the googlebot tag.
 	 *
-	 * @return array The googlebot value.
+	 * @return array The robots value with opt-in snippets.
 	 */
 	public function generate_googlebot() {
-		return [ 'max-snippet:-1', 'max-image-preview:large', 'max-video-preview:-1' ];
+		return $this->generate_snippet_opt_in();
+	}
+
+	/**
+	 * Generates the value for the bingbot tag.
+	 *
+	 * @return array The robots value with opt-in snippets.
+	 */
+	public function generate_bingbot() {
+		return $this->generate_snippet_opt_in();
+	}
+
+	/**
+	 * Generates a snippet opt-in robots value.
+	 *
+	 * @return array The googlebot value.
+	 */
+	private function generate_snippet_opt_in() {
+		if ( isset( $this->robots['index'] ) && $this->robots['index'] === 'noindex' ) {
```

Comment:
If `$this->robots` is an empty array that would equal `$this->robots['index']` being `'noindex'`.
```suggestion
		if (
			empty( $this->robots ) ||
			( isset( $this->robots['index'] ) && $this->robots['index'] === 'noindex' )
		) {
```

Reply:
Hmm not sure I follow, the `isset` would fail then, right?

Reply:
Discussed with Herre, not needed.